client/src/glossary/SideBarActivator.js

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+
+import { withRouter } from "react-router-dom";
+
+import { glossaryEntryStore } from "src/models/glossary";
+
+import { BackToTop } from "src/components";
+
+import { GlossaryMenu } from "./GlossaryMenu";
+
+const ROUTES_WITHOUT_GLOSSARY = {
+  "/start": true,
+  "/glossary": true,
+};
+
+const SidebarActivator = withRouter(
+  class SidebarActivator extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        show_sidebar: true,
+        showGlossary: false,
+        glossaryItem: null,
+        showList: true,
+      };
+
+      this.menu_ref = React.createRef();
+    }
+    itemClick = (e) => {
+      const target = e;
+      const glossary_item_key = target.dataset.ibttGlossaryKey;
+      const glossary_item = glossaryEntryStore.lookup(glossary_item_key);
+
+      this.setGlossaryItem(glossary_item.id);
+      this.showList(false);
+
+      this.toggleGlossary(true);
+    };
+
+    closeSidebar = (e) => {
+      const menu_node = this.menu_ref.current;
+      if (
+        this.state.showGlossary &&
+        menu_node &&
+        !menu_node.contains(e.target)
+      ) {
+        this.setState({ showGlossary: false });
+      }
+    };
+
+    handleWindowClick = (e) => {
+      e.target.getAttribute("data-toggle") === "tooltip"
+        ? this.itemClick(e.target)
+        : e.target.parentElement.getAttribute("data-toggle") === "tooltip"
+        ? this.itemClick(e.target.parentElement)
+        : this.closeSidebar(e);
+    };
+
+    handleWindowKeyDown = (e) => {
+      if (e.code === "Enter") {
+        this.handleWindowClick(e);
+      }
+    };
+
+    componentDidMount() {
+      window.addEventListener("click", this.handleWindowClick, {
+        capture: true,
```

Comment:
Not sure I understand your question... but the use of capture was to deal with an order of events issue in the DOM

Reply:
I was just curious because capturing phase event listeners are pretty rarely used. Was curious why it was used here (assumed maybe you saw an article recommending them for the event delegation pattern or something). What was the specific bug?

Capturing's usually not recommended. Practically, that's because it _is_ uncommon; devs will be thinking of events in terms of bubbling by default so it can be confusing when debugging down the line. More theoretically, it's not recommended because it gives priority to handlers further away from the actual target which is less logical than bubbling. 

Doesn't mean you can't use it, but maybe treat it as a "hack" and leave a comment here on why it was necessary (so that  future devs don't blindly copy the pattern or anything).